sdk/trace/src/jmh/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessorMultiThreadBenchmark.java

```diff
@@ -42,7 +42,7 @@
     private long exportedSpans;
     private long droppedSpans;
 
-    @Setup(Level.Iteration)
+    @Setup(Level.Trial)
```

Comment:
I think this will mess up with the exporter metrics..

Reply:
Since when collecting metrics it clears the current values I think it might be ok. Either way, either this needs to be `Trial` or `tearDown` needs to be `Iteration` generally we'd only want to start one BSP per trial though I think.

Reply:
could you please run this benchmark and verify the exportedSpans metric is valid, i.e; current values do get cleared after collecting them at the end of each iteration.

Reply:
It does keep on going up now - I don't think we should need to restart the BSP for it though. @jkwatson Do you know a nice way to aggregate the metrics into a rate for reporting in the JMH benchmark?

Reply:
I don't understand the question. Is something not working about the way things are before this change? Also, should we be doing a `forceFlush()` at the iteration `teardown` ?

And, honestly, I'm not sure I understand the purpose of this benchmark in the first place. I ran this on the main branch, and for the 20-thread case, we drop almost all of the spans. Is the goal to see if we can make the BSP drop fewer spans if we can improve things?

Reply:
BTW, I think we definitely start starving the export thread when we get to a lot of input threads:

![image](https://user-images.githubusercontent.com/858731/111808517-52758100-8891-11eb-9e96-f071dde871e0.png)

Reply:
(note: the jctools version is much, much better than the ABQ version on `main`):
![image](https://user-images.githubusercontent.com/858731/111808698-7a64e480-8891-11eb-964b-9e098418f276.png)

Reply:
For the metric, I think reporting dropped spans per second would be most consistent with how other numbers are also per s. I suspect otel metrics are supposed to provide a way to get such a metric :) The goal of the bench would be to export more spans per second hence wanting that (I guess "not dropping" is similar but given BSP's job is to export spans, I guess exported spans is the number more interesting).

Reply:
Though if it's annoying to export a rate, I guess total spans isn't so bad either since we know the duration of the benchmarks are fixed (just need to tweak the aux counters definition I think so jmh doesn't report it with that /s).

Reply:
> The goal of the bench would be to export more spans per second hence wanting that (I guess "not dropping" is similar but given BSP's job is to export spans

I think we should stick with exports / sec. It is natural and conveys directly what we care about. Dropped / sec is the reverse of it.

Reply:
It seems that jmh is doing the right thing here to make sure that all of the "/s" values are correct (at least they add up roughly correctly before the change here), so I think sticking with what we have now is good.

Reply:
I think this change still needs to be reverted @anuraaga

Reply:
Ok I went ahead and changed the shutdown to be per-iteration too then, need one of the changes to make sure threads are closed or JMH complains (at least to me). I don't think we actually wanted to initialize a whole BSP (worker thread, etc) per iteration though.

Reply:
What complaints did you get from jmh?

Reply:
Something about threads not all being shut down, waiting XXX seconds for them to shutdown. Which is definitely true since currently we don't call shutdown on every created BSP.

Reply:
Ah yes. I had been wondering about that. Good find!